sample_apps/rso_sample_apps/rust/src/config.rs

```diff
@@ -0,0 +1,261 @@
+use config::{Config, Environment, File};
+use log::debug;
+use serde_derive::{Deserialize, Serialize};
+
+/// The server TLS configuration
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+pub struct Tls {
+    /// The TLS certificate
+    pub cert: String,
+    /// The TLS key
+    pub key: String,
+}
+
+/// The server configuration
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+pub struct Server {
+    /// The server address
+    pub addr: String,
+    /// The server TLS configuration
+    pub tls: Option<Tls>,
+}
+
+/// Configuration holds the  required parameters to connect to a Git provider.
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+pub struct Configuration {
+    /// The  client ID.
+    pub client_id: String,
+    /// The client secret.
+    pub client_secret: String,
+    /// The provider URL.
+    pub provider_url: String,
+    /// The server configuration.
+    pub server: Server,
+    /// The callback host.
+    pub callback_host: String,
+    /// The RiotGamesAPI Token
+    pub api_token: String,
+    /// The account data URL
+    pub account_data_url: String,
+    /// The champion data URL
+    pub champion_data_url: String,
+}
+
+impl Configuration {
+    /// The callback URL that the OAuth provider will redirect to after authentication.
+    pub fn callback_url(&self) -> String {
+        let protocol = match self.server.tls {
+            Some(_) => "https://",
+            None => "http://",
+        };
+        format!("{}{}/oauth", protocol, self.callback_host)
+    }
+
+    /// Returns the provide token endpoint.
+    pub fn token_url(&self) -> String {
+        format!("{}/token", self.provider_url)
+    }
+
+    /// Returns the URL of the OAuth authorize endpoint.
```

Comment:
Rust supports markdown, we can improve comments with a better struct, for instance: 

/// Returns the OAuth authorization code.
///
/// The authorize endpoint is where the user is redirected to grant the application permission to access their data.
/// The user will be prompted to log in to their account  and grant the application the requested permissions. Once the user has granted the permissions, they will be redirected back to the application with an authorization code.
///
/// # Returns
///
/// A `String` containing the authorization code.
///